src/autogen.sh

```diff
@@ -1,10 +1,3 @@
 #!/bin/sh
 rm -rf autom4te.cache
-AUTOGEN_TARGET=${AUTOGEN_TARGET-configure:config.h.in}
-set -e
-case :$AUTOGEN_TARGET: in
-*:configure:*) autoconf; touch configure ;;
-esac
-case :$AUTOGEN_TARGET: in
-*:config.h.in:*) autoheader; touch config.h.in ;;
```

Comment:
I need to check why we had both of these before, I think it has something to do work make invoking this script itself

Reply:
>I need to check why

Please, because It wasn't obvious for me and I chose simplicity over complexity.

Reply:
```
ifeq ($(TRIVIAL_BUILD),no)
configure: configure.ac
        AUTOGEN_TARGET=configure ./autogen.sh

config.h.in: configure.ac
        AUTOGEN_TARGET=config.h.in ./autogen.sh
```

I haven't been [comfortable with make rules that generate multiple targets](https://www.gnu.org/software/automake/manual/html_node/Multiple-Outputs.html)—read the docs about it, they're a hoot of workarounds piled on workarounds, so I allowed make to invoke it in this way. It ensures that each file is generated by a distinct invocation of autogen.sh. With your change,

 * autogen.sh will be invoked twice
 * to update all the same files
 * possibly in parallel

On the other hand, if the stanza for `config.h.in` isn't present, then `make` may believe it does not need to regenerate `config.h` from `config.h.in`:
```
config.h: config.h.in config.status
        @./config.status -q --header=$@                                         
```
... depending when it stats config.h.in.

I have to admit I am not confident whether this works as intended or not, users should just re-run `autogen.sh` manually when modifying configure.ac. But I'd rip out some amount of Makefile goo, trying to make it clear that make can't really re-invoke configure and restart itself as it actually needs to do.

Reply:
```make
config.h.in: configure.ac
	AUTOGEN_TARGET=config.h.in ./autogen.sh
#…
Makefile: config.h  
```

I've been trying to disentangle this.

Here's what I think I see:
 * There are a lot of places where `Makefile` is specifically depended. Since it was renamed to `makefile` all these explicit dependencies need to be revised. (on left and right sides)
 * There's special machinery in make where, if a file used as input by make is out of date, make will regenerate it and then reinvoke itself. This seems to allow getting rid of the special `config.h: ...` rule that I noticed at first, because make will re-invoke itself as long as there's a correct chain from `configure.ac` to `[Mm]akefile` (whatever filename is being used)
 * leaving the rule is bad, because it'll invoke the `autogen` script twice in parallel to generate the same files. Who knows what you end up with in this case

It's unfortunate that build system bugs of the type I'm describing above are often hard to reproduce and can depend on which jobs are invoked in what order, the job parallelism, etc.  The problem of renaming Makefile but not changing the rules that refer to it could easily go unnoticed, because it doesn't produce any overt behavior most of the time -- you just get rebuilds that are incomplete.